spec/fixtures/mocks/lua-resty-dns/resty/dns/resolver.lua

```diff
@@ -101,14 +101,14 @@ resolver.query = function(self, name, options, tries)
   end
 end
 
-do
-  local semaphore = require "ngx.semaphore"
-  local old_post = semaphore.post
-  function semaphore.post(self, n)
-    old_post(self, n)
-    ngx.sleep(0)
-  end
-end
+-- do
+--   local semaphore = require "ngx.semaphore"
+--   local old_post = semaphore.post
+--   function semaphore.post(self, n)
+--     old_post(self, n)
+--     ngx.sleep(0)
+--   end
+-- end
```

Comment:
https://github.com/Kong/kong/pull/7704

This patch looks like it's meant to fix some OpenResty bug, and nothing tested wrong when I disabled it. It doesn't seem to be timer-related.

Reply:
This was a fix for a lot of flaky tests. @javierguerragiraldez might know better if this can be removed or not.

Here's the underlying issue: https://github.com/openresty/lua-resty-core/issues/337

Reply:
that "some openresty bug" is actually in the docs (https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/semaphore.md#post):
> Releases n (default to 1) "resources" to the semaphore instance.
This will not yield the current running "light thread".

but most code seems to assume that yield, and so we get a long runs before the woken-up thread gets some CPU time.  if the new time library does include that yield, then yes, this one should be removed.  if not, then it does help a lot.

Reply:
that is not the bug, I would expect the current coro to NOT yield. That is fine. The problem is that if there is nothing else to do, OpenResty simply exits without EVER resuming the released thread.